### PR TITLE
 fix bug: test storage won't be updated when a test file is deleted

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

File: extension/src/extension.ts [hunk 1]
```diff
@@ -76,6 +66,14 @@ export async function activate(context: ExtensionContext) {
         // onDidChange.fire();
     });
 
+    workspace.onDidCloseTextDocument((document) => {
+        const uri = document.uri;
+        // if a test file is removed, should update test storage.
+        if (!fs.existsSync(uri.fsPath)) {
+            testResourceManager.storeTests(uri, undefined);
```
Comment (jdneo) [inline extension/src/extension.ts]: remove tests in the method named `storeTests` is confusing. Can we make another method named something like `removeTests()`?

Comment (ansyral) [reply extension/src/extension.ts]: make sense

File: extension/src/extension.ts [hunk 2]
```diff
@@ -59,26 +59,14 @@ export async function activate(context: ExtensionContext) {
...
-        const uri = document.uri;
-        // if a test file is removed, should update test storage.
-        if (!fs.existsSync(uri.fsPath)) {
-            testResourceManager.removeTests(uri);
-        }
-    });
-
-    workspace.onDidSaveTextDocument((document) => {
-        const uri = document.uri;
+    const watcher = workspace.createFileSystemWatcher('**/*.{[jJ][aA][vV][aA]}');
```
Comment (jdneo) [inline extension/src/extension.ts]: Looks like we can push the `watcher` into `context.subscription`?

https://github.com/Microsoft/vscode/blob/80d6fa0b8eb5f4c0959b9e2925ec7fd28692880f/src/vs/vscode.d.ts#L1490
```javascript
export interface FileSystemWatcher extends Disposable
```

Comment (ansyral) [reply extension/src/extension.ts]: yes, we should!

**Pull request merged**